src/dune_engine/fs_memo.ml

```diff
@@ -2,9 +2,13 @@ open! Stdune
 open! Import
 open Memo.Build.O
 
+type watch =
```

Comment:
I think I prefer the naming used in Jenga: `Via_parent` and `Directly`.

Reply:
(And `watch` is named `how`.)

Reply:
Not sure I like `how`. For short names like that, the module name should give context but `Fs_memo.how` is equally opaque.

Reply:
Agreed, `watch` seems fine.